core/lib/zksync_core/src/fee_model.rs

```diff
@@ -0,0 +1,126 @@
+use std::{fmt, sync::Arc};
+
+use zksync_types::{
+    fee_model::{
+        BatchFeeInput, L1PeggedBatchFeeModelInput, MainNodeFeeModelConfig, MainNodeFeeParams,
+        PubdataIndependentBatchFeeModelInput,
+    },
+    U256,
+};
+
+use crate::l1_gas_price::L1GasPriceProvider;
+
+/// Trait responsiblef for providign fee info for a batch
+pub trait BatchFeeModelInputProvider: fmt::Debug + 'static + Send + Sync {
+    fn get_batch_fee_input(&self, scale_l1_prices: bool) -> BatchFeeInput {
```

Comment:
`scale_l1_prices` is not really self-explonatory...

Reply:
I'll add a comment